daemon/containerd/image_snapshot.go

```diff
@@ -67,3 +70,41 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 	_, err = s.Prepare(ctx, id, parent)
 	return err
 }
+
+// calculateSnapshotParentUsage returns the usage of all ancestors of the
+// provided snapshot. It doesn't include the size of the snapshot itself.
+func calculateSnapshotParentUsage(ctx context.Context, snapshotter snapshots.Snapshotter, snapshotID string) (snapshots.Usage, error) {
+	info, err := snapshotter.Stat(ctx, snapshotID)
+	if err != nil {
+		return snapshots.Usage{}, errors.Wrapf(err, "snapshotter.Stat failed for %s", snapshotID)
```

Comment:
Should this be considered an `errdefs.System` ? Or should we apply that at the call-site ?

Reply:
Yeah, makes sense to do it here. Done, thanks!